crashy/lib/main.dart

```diff
@@ -0,0 +1,171 @@
+  if (isInDebugMode) {
+    print(stackTrace);
+    print('In dev mode. Not sending report to Sentry.io.');
+    return;
+  }
+
+  print('Reporting to Sentry.io...');
+
+  final SentryResponse response = await _sentry.captureException(
+    exception: error,
+    stackTrace: stackTrace,
+  );
+
+  if (response.isSuccessful) {
+    print('Success! Event ID: ${response.eventId}');
+  } else {
+    print('Failed to report to Sentry.io: ${response.error}');
+  }
+}
+
+Future<Null> main() async {
+  // This captures errors reported by the Flutter framework.
+  FlutterError.onError = (FlutterErrorDetails details) async {
+    if (isInDebugMode) {
+      // In development mode simply print to console.
+      FlutterError.dumpErrorToConsole(details);
+    } else {
+      // In production mode report to the application zone to report to
+      // Sentry.
+      Zone.current.handleUncaughtError(details.exception, details.stack);
+    }
+  };
+
+  // This creates a [Zone] that contains the Flutter application and stablishes
+  // an error handler that captures errors and reports them.
+  //
+  // Using a zone makes sure that as many errors as possible are captured,
+  // including those thrown from [Timer]s, microtasks, I/O, and those forwarded
+  // from the `FlutterError` handler.
+  //
+  // More about zones:
+  //
+  // - https://api.dartlang.org/stable/1.24.2/dart-async/Zone-class.html
+  // - https://www.dartlang.org/articles/libraries/zones
+  runZoned<Future<Null>>(() async {
+    runApp(new CrashyApp());
+  }, onError: (error, stackTrace) async {
+    await _reportError(error, stackTrace);
+  });
+}
+
+class CrashyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Crashy',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyHomePage(),
+    );
+  }
+}
+
+class MyHomePage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text('Crashy'),
+      ),
+      body: new Center(
+        child: new Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            new RaisedButton(
+              child: new Text('Dart exception'),
+              elevation: 1.0,
+              onPressed: () {
+                throw new StateError('This is a Dart exception.');
+              },
+            ),
+            new RaisedButton(
+              child: new Text('async Dart exception'),
+              elevation: 1.0,
+              onPressed: () async {
+                foo() async {
+                  throw new StateError('This is an async Dart exception.');
+                }
+                bar() async {
+                  await foo();
+                }
+                await bar();
+              },
+            ),
+            new RaisedButton(
+              child: new Text('Java exception'),
```

Comment:
"Method channel" might be better here, unless this doesn't work on iOS.

Reply:
If it's OK with you, I may punt on this suggestion for the repo refactor, and file an issue to do the same, just to separate concerns here.